application-workloads/sccm/sccm-currentbranch/main.bicep

```diff
@@ -0,0 +1,250 @@
+@secure()
+param _artifactsLocationSasToken string = ''
+
+@description('Location for all resources.')
+param location string = resourceGroup().location
+
+@description('SQL VM size')
+param SQLVmSize string = 'Standard_B2ms'
+
+var dscScript = 'DSC/DCConfiguration.zip'
+var virtualNetworkName = '${toLower(prefix)}-vnet'
+var domainName = 'contoso.com'
+var timeZone = 'UTC'
+var networkSettings = {
+  virtualNetworkAddressPrefix: '10.0.0.0/16'
+  subnetAddressPrefix: '10.0.0.0/24'
+  virtualMachinesIPAddress: '10.0.0.'
+  subnetRef: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetworkName, 'default')
+  privateIPAllocationMethod: 'Static'
+  publicIpAllocationMethod: 'Dynamic'
+}
+var securityGroupRuleName = 'default-allow-rdp'
+var securityGroupRule = {
+  priority: 1000
+  sourceAddressPrefix: '*'
+  protocol: 'Tcp'
+  destinationPortRange: '3389'
+  access: 'Allow'
+  direction: 'Inbound'
+  sourcePortRange: '*'
+  destinationAddressPrefix: '*'
+}
+var vmrole = [
+  'DC'
+  'DPMP'
+]
+var vmInfo = {
+  DC: {
+    name: 'DC01'
+    disktype: 'Premium_LRS'
+    Size: vmSize
+    imageReference: {
+      publisher: 'MicrosoftWindowsServer'
+      offer: 'WindowsServer'
+      sku: '2022-datacenter'
+      version: 'latest'
+    }
+  }
+  DPMP: {
+    name: 'DPMP01'
+    disktype: 'Premium_LRS'
+    size: vmSize
+    imageReference: {
+      publisher: 'MicrosoftWindowsServer'
+      offer: 'WindowsServer'
+      sku: '2022-datacenter'
+      version: 'latest'
+    }
+  }
+}
+var siteRole = ((configuration == 'Standalone') ? [
+  'PS'
+] : [
+  'CS'
+  'PS'
+])
+var siteInfo = ((configuration == 'Standalone') ? json('{"PS":{"name":"PS01","DiskType":"Premium_LRS","size":"${SQLVmSize}","imageReference":{"publisher": "MicrosoftSQLServer","offer": "SQL2019-WS2019","sku": "Standard","version": "latest"}}}') : json('{"CS":{"name":"CS01","DiskType":"Premium_LRS","size":"${SQLVmSize}","imageReference":{"publisher": "MicrosoftSQLServer","offer": "SQL2019-WS2019","sku": "Standard","version": "latest"}},"PS":{"name":"PS01","DiskType":"Premium_LRS","size":"${SQLVmSize}","imageReference":{"publisher": "MicrosoftSQLServer","offer": "SQL2019-WS2019","sku": "Standard","version": "latest"}}}'))
+var ClientName = [for i in range(0, ClientsCount): '${prefix}Cl0${(i + 1)}']
+var clientRole = [for i in range(0, ClientsCount): 'Client${(i + 1)}']
+var clientInfo = [for i in range(0, ClientsCount): json('{"Client${(i + 1)}":{"name": "Cl0${(i + 1)}","disktype": "Premium_LRS","size": "${vmSize}","imageReference": {"publisher": "MicrosoftWindowsDesktop","offer": "Windows-11","sku": "win11-23h2-ent","version": "latest"}}}')]
+
+resource prefix_vmInfo_vmInfo_vmRole_name_vmInfo_siteInfo_clientInfo_vmInfo_siteInfo_clientRole_vmInfo_siteInfo_name_siteInfo_siteRole_vmInfo_name 'Microsoft.Compute/virtualMachines@2019-12-01' = [for i in range(0, ((length(vmInfo) + length(siteInfo)) + length(clientRole))): {
```

Comment:
can we update the resource name to something coherent? it looks like this was autogenerated by the decompiler.